src/panels/edit-panel/panelCtrl.ts

```diff
@@ -84,11 +83,23 @@ export class PlotlyPanelCtrl extends MetricsPanelCtrl {
   async onUpload(net: any) {
     this.panel.predictionSettings.json = JSON.stringify(net);
     this.publishAppEvent(AppEvents.alertSuccess, ['File Json Caricato']);
+    this.panel.predictionSettings.predittori = net.Predictors.map((a, index) => {
+      return { id: index, name: a };
+    });
   }
 
   async delete_json_click() {
     this.predictionPanelConfig.json = null;
     this.publishAppEvent(AppEvents.alertSuccess, ['File Json Cancellato']);
+    this.panel.predictionSettings.predittori = null;
+  }
+
+  async update_queries() {
+    // per influxDB il nome del campo potrebbe essere name anziché alias
+    this.panel.predictionSettings.query = this.queryList.map(a => {
+      return { id: a.refId, name: a.alias };
+    });
+    // comportamento anomalo: cancella la nodeMap
```

Comment:
prova a mettere così

```
this.panel.predictionSettings.query = _.clone(this.queryList.map(a => {
      return { id: a.refId, name: a.alias };
    }));
```

Reply:
Questa modifica + this.refresh() all'interno di update_query non hanno fatto differenza

Reply:
Edit: la refresh fa danni, con la refresh viene aggiornato anche il set di dati, che causa a sua volta una nuova chiamata dell'onDataReceived, andando in loop infinito